iStore/Home/HomeViewModel.swift

```diff
@@ -19,24 +18,22 @@ final class HomeViewModel: ObservableObject {
     }
     
     func getProducts() {
-        
         let url = allProductsUrl
         
         networkManager.loadData(withURL: url) { [weak self] (result: Result<ProductsData,NetworkError>) in
-            
             guard let self = self else {
                 return
             }
             
             switch result {
-            case .success(let data):
-                DispatchQueue.main.async {
-                    self.products.append(contentsOf: data.products)
-                    self.isloading = false
-                }
-                
-            case .failure(let error):
-                print(NetworkError.unknownError(error))
+                case .success(let data):
+                    DispatchQueue.main.async {
```

Comment:
Why are you using main queue here?

Reply:
Because It updates the UI.

UI updates should be on the main thread.

Reply:
But you are in the view model. Why do you want to update the UI from the view model?

Reply:
why wouldn't I ?
Isn't the viewModel meant for removing the logic the view ?

Reply:
You should do all logic in view model as you said. When you want to refresh the view and use the main thread, you should make it in the view not view model.